### DISCUSSION TITLE
Is there any way to know if the viewer has loaded already a specific zoom level on a terrain?

### DISCUSSION THREAD

**Discussion by carlosdt1996** [pelicanmapping/osgearth#2435] [Support Q&A]
Hi everyone, I would like to know if there is any callback or boolean implemented to know when the viewer has loaded the zoom needed when I move the camera to a specific location, I would need to know exactly when it loads through code, thanks!

**Comment by gwaldron** [comment]
There is not. This is something people have requested in the past so it's on the list.

**Reply by carlosdt1996** [reply]
Could you give me any hint on where could I implement it? I have been looking into the code and I wasn´t able find anything related.

**Reply by gwaldron** [reply]
Not really as this is a fairly complex feature to implement. Like I said it is on the roadmap.

In the meantime, if you can detail the reasons behind your request, it would be helpful in designing a solution. And maybe there's another way to accomplish what you want.

**Reply by carlosdt1996** [reply]
To summarize I am working on a software to orthorrectify photos on the terrain, so in order to do the process I need to know exactly when the tile nodes where the camera is pointing at have the terrain loaded with the maximum level of detail. This is a script that works by its own so I need to know when those are loaded through a callback, right now we are doing a "sleep" pushing like 10k frames to be sure that the terrain is loaded but this make the software pretty slow. Thanks for your time.
